### ISSUE TITLE
Running npm install on jhipster itself fails on windows

### ISSUE THREAD

**Issue by naris** [jhipster/generator-jhipster#19981]
##### **Overview of the issue**

When you clone the  [jhipster project](https://github.com/jhipster/generator-jhipster.git) on a windows machine and then run npm install, it fails. npm link and npm ci also fails. The official windows build is also in a failed state 

npm install fails with
```
> generator-jhipster@7.9.3 copy-files
> cpy 'jdl/*.json' 'dist/jdl' && cpy 'generators/**' '!**/*.(js|cjs|mjs|ts|cts|mts|snap)' '!**/__*/*' dist/generators

'cjs' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code 255
npm ERR! path C:\src\generator-jhipster
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c npm run build
```

I have also seen this
```
> generator-jhipster@7.9.3 copy-files
> cpy 'generators/**' '!**/*.(js|ts|cjs|mjs|snap)' '!**/__*/*' dist/generators && cpy 'package.json' dist

'ts' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code 255
npm ERR! path C:\src\generator-jhipster
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c npm run build
```

##### **Motivation for or Use Case**

This makes it difficult to use the pre-release version of jhipster and blueprints that depend on it (such as the new version of yelhouti's primeng blueprint, or contribute to, or debug, jhipster on a windows machine :(

##### **Reproduce the error**

Clone the [jhipster project](https://github.com/jhipster/generator-jhipster.git) on a windows machine and then run npm install

##### **Related issues**

No, but the official windows build is currently failing with a similar issue.

##### **JHipster Version(s)**

main branch and [yelhoutie's composite-key branch](https://github.com/yelhouti/generator-jhipster/tree/composite-key)

##### **Browsers and Operating System**

Windows 10 Enterprise 21H2 19044.2006

- [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by venkatnaveen7** [comment]
Im also facing same
'cjs' is not recognized as an internal or external command,
operable program or batch file.

**Comment by kaidohallik** [comment]
Fixed by #20330
